null_safety/calculate_lix/lib/lix.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+class Lix {
+  // Based on:
+  // https://readabilityformulas.com/the-LIX-readability-formula.php.
+  int words; // Number of words in general.
+
+  int longWords; // Number of words with more than 6 characters.
+
+  int periods; // Number of periods.
+
+  // TIP: `readability` isn't passed to the constructor, but calculated. By
+  // adding the late keyword we tell the analyzer that we've handled
+  // initializing it.
+  late int readability; // The lix readability index calculated.
+
+  // TIP: Because the fields are all non-nullable, the constructor must
+  // initialize them all. We can either set a default value, or like here make
+  // the fields 'required', meaning that a value must be passed to the
+  // constructor. 'required' is a new keyword introduced as part of null safety
+  // which replaces the previous '@required' annotatation.
```

Comment:
Is it worth calling out again that the `late` fields don't need to be initialized here, even though they are also non-nullable?

Reply:
I think I'll leave that out from here for simplicity